### PR TITLE
Steven P01

### PR DESCRIPTION
**Pull request by Steven-Pinkerton**

### CODE REVIEW

File: src/Progression/P01.hs [hunk 1]
```diff
@@ -29,5 +29,5 @@ main :: IO ()
 main = do
   let name :: String = "Joe"
       greeting :: String = "Hello, "
-      message = undefined
+      message = greeting ++ name
```
Comment (srid) [inline src/Progression/P01.hs:32]: You can also use `<>` here. This is coming from the `Semigroup` class. It works with any type with that instance, not just lists (which is what `++` works on).

Comment (Steven-Pinkerton) [reply src/Progression/P01.hs]: Understood changed.

Comment (srid) [reply src/Progression/P01.hs]: Don't forget to read about it if you are curious :-)

- https://en.wikipedia.org/wiki/Semigroup
- https://livebook.manning.com/book/get-programming-with-haskell/chapter-17/
- https://typeclasses.com/semigroup
- https://serokell.io/blog/whats-that-typeclass-monoid

File: src/Progression/P01.hs [hunk 2]
```diff
@@ -25,9 +25,36 @@ program to your instructor.
 -}
 module Progression.P01 where
 
-main :: IO ()
-main = do
-  let name :: String = "Joe"
+
+printGreeting' :: String -> IO ()
```
Comment (srid) [inline src/Progression/P01.hs:29]: Naming could be more useful, eg: `greetPerson`

File: src/Progression/P01.hs [hunk 3]
```diff
@@ -25,9 +25,36 @@ program to your instructor.
...
+printGreeting' x = 
+  putStrLn ("Hello, " <> x)
```
Comment (srid) [inline src/Progression/P01.hs:31]: You can also write this using `$`:

```haskell
putStrLn $ "Hello, " <> x
```

**Review by srid — APPROVED** [review]

**Pull request closed without merging**